modules/generator/processor/servicegraphs/servicegraphs.go

```diff
@@ -190,6 +190,17 @@ func (p *Processor) consume(resourceSpans []*v1_trace.ResourceSpans) (err error)
 							e.ServerService = dbName
 							e.ServerLatencySec = spanDurationSec(span)
 						}
+
+						if p.Cfg.EnableExtraUninstrumentedServicesLabels {
+							switch e.ConnectionType {
```

Comment:
this is mutually exclusive, but is it possible that  a service can be both database and messaging system? eg Redis

Reply:
it is - because at this point it has been decided what the service is acting as, depending on the presence/absence of the `db.name` resource attribute